src/client.jl

```diff
@@ -599,4 +600,49 @@ is_request_interrupted(ex::TaskFailedException) = is_request_interrupted(ex.task
 is_request_interrupted(ex::CompositeException) = any(is_request_interrupted, ex.exceptions)
 is_request_interrupted(ex::InvocationException) = ex.reason == "request was interrupted"
 
+
+"""
+    deserialize_file(api_call::Function;
+        folder_path::String=pwd(),
+        rename_file::String="",
+        overwrite::Bool=true
+        )::Tuple{Any,ApiResponse,String}
+
+    Saves response body into a file in a temporary folder,
+    using the filename from the `Content-Disposition` header if provided.
+    - `api_call`: API function that return `(result, http_response)` Tuple.
+    - `folder_path`: file save location, default value is `pwd()``.
+    - `rename_file`: rename the file, default value is `""`.
+    - `overwrite`: overwrite file if already exist, default is `true`.
+    - return: (result, http_response, file_path).
+"""
+function deserialize_file(api_call::Function;
+    folder_path::String=pwd(),
+    rename_file::String="",
+    overwrite::Bool=true
+    )::Tuple{Any,ApiResponse,String}
+    
+    result, http_response = api_call()
+
+    content_disposition_str = OpenAPI.Clients.header(http_response.raw,"content-disposition","")
+    content_type_str = OpenAPI.Clients.header(http_response.raw,"content-type","")
+    
+    file_name = if length(rename_file) > 0
+        rename_file
+    elseif length(content_disposition_str) > 0
+        content_disposition_str
+    else 
+        "response"*extension_from_mime(MIME(content_type_str))
+    end
+
+    file_path = joinpath(mkpath(folder_path),file_name)
+
+    if !(isfile(file_path)&&!overwrite)
```

Comment:
Not sure if we should accept an option to not `overwrite`. There may be more than one way the method may be expected to behave when file already exists and overwrite is false. Same goes for the folder name.

I feel it is best left for the caller to pass the exact folder and file names, and let the method always replace the file.